### PR TITLE
`Development:` Add JavaDocs to UtilServices and Factories [8]

### PR DESCRIPTION
**Pull request by laadvo**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to helper methods.

### Description

Added JavaDoc to following files:

- [x] exercise/quizexercise/QuizExerciseFactory.java
- [x] exercise/quizexercise/QuizExerciseUtilService.java

also moved some helper methods from `QuizExerciseUtilService` to `QuizExerciseFactory`.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unchanged

### CODE REVIEW

**Review by MaximilianAnzinger — DISMISSED** [review]

File: src/test/java/de/tum/in/www1/artemis/exercise/quizexercise/QuizExerciseFactory.java [hunk 1]
```diff
@@ -45,9 +62,10 @@ public static QuizExercise createQuizForExam(ExerciseGroup exerciseGroup) {
     }
 
     /**
-     * initializes a quiz with all different types of questions
+     * Initializes a quiz exercise with different types of questions.
+     * After initialization, the quiz consists of one multiple choice, one drag and drop, and one short answer question.
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/exercise/quizexercise/QuizExerciseFactory.java:66]: This is only true if the Quiz did not contain Questions before. The method adds questions but does not remove potentially existing ones. Maybe update the comment or remove existing questions before adding the new ones.

File: src/test/java/de/tum/in/www1/artemis/exercise/quizexercise/QuizExerciseFactory.java [hunk 2]
```diff
@@ -293,41 +367,55 @@ public static QuizExercise generateQuizExerciseForExam(ExerciseGroup exerciseGro
...
+     */
     public static QuizExercise generateQuizExerciseForExam(ExerciseGroup exerciseGroup, String title) {
         var quizExercise = generateQuizExerciseForExam(exerciseGroup);
         quizExercise.setTitle(title + UUID.randomUUID().toString().substring(0, 3));
         return quizExercise;
     }
 
+    /**
+     * Initializes a quiz exercise with all different types of questions.
+     * After initialization, the quiz consists of one multiple choice, one drag and drop, one short answer, and single choice question.
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/exercise/quizexercise/QuizExerciseFactory.java:390]: Same as above.

**Review by DominikRemo — DISMISSED** [review]
LGTM, just two optional nitpicks

**Review by egekurt123 — DISMISSED** [review]
Documentation text looks decent.

**Review by DominikRemo — APPROVED** [review]

**Review by egekurt123 — APPROVED** [review]

**Pull request merged**